test/src/androidTest/kotlin/io/realm/shared/notifications/RealmObjectNotificationsTests.kt

```diff
@@ -0,0 +1,179 @@
+            }
+            val observer = async {
+                obj.observe().collect {
+                    c.trySend(it)
+                }
+            }
+            assertEquals("Foo", c.receive()!!.stringField)
+            obj.update {
+                stringField = "Bar"
+            }
+            assertEquals("Bar", c.receive()!!.stringField)
+            obj.update {
+                stringField = "Baz"
+            }
+            assertEquals("Baz", c.receive()!!.stringField)
+            observer.cancel()
+            c.close()
+        }
+    }
+
+    @Test
+    override fun cancelObserve() {
+        runBlocking {
+            val obj: Sample = realm.write {
+                copyToRealm(Sample().apply { stringField = "Foo" })
+            }
+            val c1 = Channel<Sample?>(1)
+            val c2 = Channel<Sample?>(1)
+            val observer1 = async {
+                obj.observe().collect {
+                    c1.trySend(it)
+                }
+            }
+            val observer2 = async {
+                obj.observe().collect {
+                    c2.trySend(it)
+                }
+            }
+            obj.update {
+                stringField = "Bar"
+            }
+            assertEquals("Bar", c1.receive()!!.stringField)
+            assertEquals("Bar", c2.receive()!!.stringField)
+            observer1.cancel()
+            obj.update {
+                stringField = "Baz"
+            }
+            assertEquals("Baz", c2.receive()!!.stringField)
+            assertTrue(c1.isEmpty)
+            observer2.cancel()
+            c1.close()
+            c2.close()
+        }
+    }
+
+    @Test
+    override fun initialElement() {
+        runBlocking {
+            val c = Channel<Sample?>(1)
+            val obj = realm.write {
+                copyToRealm(
+                    Sample().apply {
+                        stringField = "Foo"
+                    }
+                )
+            }
+            val observer = async {
+                obj.observe().collect {
+                    c.trySend(it)
+                }
+            }
+            assertEquals("Foo", c.receive()!!.stringField)
+            observer.cancel()
+            c.close()
+        }
+    }
+
+    @Test
+    override fun deleteObservable() {
+        runBlocking {
+            val c = Channel<Sample?>(1)
+            val obj: Sample = realm.write {
+                copyToRealm(Sample().apply { stringField = "Foo" })
+            }
+            val observer = async {
+                obj.observe().collect {
+                    c.trySend(it)
+                }
+                // Emit sentinel value to signal that flow completed
+                c.send(Sample())
+            }
+            assertNotNull(c.receive())
+            realm.write {
+                delete(findLatest(obj)!!)
+            }
+            assertNull(c.receive()) // Null is sent when object is deleted
```

Comment:
Shouldn't we close/complete the Flow if the element we're observing is deleted?

Reply:
Yes, we are also doing that. But there is an open question if we should just complete the Flow directly or emit `null` before doing it, to make sure that people actually handle the case.

Otherwise they need to remember to do something like:

```
scope.launch {
                obj.observe()
                    .onCompletion { 
                        // Object was deleted
                    }
                    .collect {
                        // Handle object update
                }
}
```

Right now I opted for emitting `null`, but that is something we need to agree on. Do you have any preference?